### PR TITLE
Update Beacon/Validator Flags to Disable Reward/Cross-Link Checking by Default

### PR DESCRIPTION
**Pull request by rauljordan**
This is a follow up to #588.

---

# Description

This code fixes the commented code from #588 by using cli flags to disable certain behavior by default, such as cross-link and reward checking for validators. The idea is that we should have certain production behaviors disabled by default for people to run a local demo until these features are robust enough for users to run. This is also part of #485.

### CODE REVIEW

File: beacon-chain/utils/flags.go [hunk 1]
```diff
@@ -55,4 +54,29 @@ var (
 		Name:  "genesis-json",
 		Usage: "Beacon node will bootstrap genesis state defined in genesis.json",
 	}
+	// EnablePOWChain tells the beacon node to use a real web3 endpoint.
```
Comment (terenc3t) [inline beacon-chain/utils/flags.go]: Should we mention this is enabled or disabled by default?

File: beacon-chain/utils/flags.go [hunk 2]
```diff
@@ -55,4 +54,29 @@ var (
...
+	EnablePOWChain = cli.BoolFlag{
+		Name:  "enable-powchain",
+		Usage: "Enable a real, web3 proof-of-work chain endpoint in the beacon node",
```
Comment (terenc3t) [inline beacon-chain/utils/flags.go]: Not sure if this `Enable` should be capitalized. It's all lower cases for flags below

File: beacon-chain/blockchain/service.go [hunk 3]
```diff
@@ -106,10 +116,7 @@ func (c *ChainService) Stop() error {
 // CurrentBeaconSlot based on the seconds since genesis.
 func (c *ChainService) CurrentBeaconSlot() uint64 {
 	secondsSinceGenesis := time.Since(c.genesisTimestamp).Seconds()
-	if math.Floor(secondsSinceGenesis/8.0)-1 < 0 {
-		return 0
-	}
-	return uint64(math.Floor(secondsSinceGenesis/8.0)) - 1
+	return uint64(math.Floor(secondsSinceGenesis / 8.0))
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/service.go]: Is this change intentional?

Comment (rauljordan) [reply beacon-chain/blockchain/service.go]: oh wow I didn't add this

File: beacon-chain/simulator/service.go [hunk 4]
```diff
@@ -43,13 +44,13 @@ type Config struct {
...
 	BeaconDB        ethdb.Database
-	DevMode         bool
+	EnablePOWChain  bool
 }
 
 // DefaultConfig options for the simulator.
 func DefaultConfig() *Config {
 	return &Config{
-		Delay:           time.Second * time.Duration(params.GetConfig().SlotDuration),
+		Delay:           time.Second * 5,
```
Comment (prestonvanloon) [inline beacon-chain/simulator/service.go]: Should this be changed? It looks like it was based on the slot duration?

File: validator/beacon/service_test.go [hunk 5]
```diff
@@ -325,7 +356,7 @@ func TestListenForAssignmentProposer(t *testing.T) {
 		gomock.Any(),
 	).Return(stream, nil)
 	b.cancel()
-
+	//
```
Comment (prestonvanloon) [inline validator/beacon/service_test.go]: Revert this one
